### PR TITLE
[DEVOPS-1750] Build and release pipeline for go SDK

### PR DESCRIPTION
**Pull request by michalchecinski**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

- **languages/go/.version:** Add file to hold current go SDK version
- **.github/workflows/version-bump.yml** Add go SDK to version bump workflow
- **.github/workflows/golang-release.yml** replace it with `.github/workflows/release-go.yaml` workflow.
- **.github/workflows/build-go.yaml** Add build go as a separate workflow
- **.github/workflows/release-go.yml** Update release go pipeline to our standards. Sync go SDK folder to external repo. Create release tag.

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

File: .github/workflows/build-go.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - hotfix-rc
+
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    needs:
+      - build_rust
+      - generate-schemas
```
Comment (mimartin12) [inline .github/workflows/build-go.yaml:15]: ⚠️  Dependencies should be updated, I don't see these being used in this workflow.

File: .github/workflows/build-go.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Setup Go environment
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: ${{ env.GO_VERSION }}
```
Comment (mimartin12) [inline .github/workflows/build-go.yaml:24]: ❓  How does `${{ env.GO_VERSION }}` get it's value?

**Review by mimartin12 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-go.yaml (hunk 2 above)
Comment (michalchecinski) [reply .github/workflows/build-go.yaml:25]: Fixed.

File: .github/workflows/build-go.yaml (hunk 1 above)
Comment (michalchecinski) [reply .github/workflows/build-go.yaml]: Fixed.

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**